federationapi/routing/invite.go

```diff
@@ -38,7 +38,17 @@ func InviteV2(
 	rsAPI api.RoomserverInternalAPI,
 	keys gomatrixserverlib.JSONVerifier,
 ) util.JSONResponse {
+
 	inviteReq := gomatrixserverlib.InviteV2Request{}
+	// Check to see if the room_version is supported
+	if _, err := roomserverVersion.SupportedRoomVersion(inviteReq.RoomVersion()); err != nil {
```

Comment:
`inviteReq` hasn't been unmarshalled yet, so this doesn't make sense. It has to happen after `json.Unmarshal(request.Content(), &inviteReq)`.

Reply:
Fixed